### ISSUE TITLE
can't use Enum in unions yet

### ISSUE THREAD

**Issue by rmosolgo** [crystal-lang/crystal#2732]
Hi, thanks again for all your work on Crystal! 

I'm hoping to write an object like this one, where a value in the attributes hash may be an Enum: 

```crystal
class Entity 
  alias AttributeName = String 
  alias AttributeValue = String | Enum # | Bool | ... 
  alias Attributes = Hash(AttributeName, AttributeValue) 

  def initialize 
	@attributes = Attributes.new 
  end 

  def set(attr_name, attr_value) 
    @attributes[attr_name] = attr_value 
  end 

  def get?(attr_name)
    @attributes[attr_name]? 
  end 
end 
```

However, it fails with: 

```
Error in line 3: can't use Enum in unions yet, use a more specific type
```

https://carc.in/#/r/10ng

- Is there another way to accomplish this? 
- Is there another issue to track progress on this behavior? (If so, feel free to close this one)
- How "fixable" is this? (If I understood the problem, could I contribute to the solution? Or is it a "big picture"-type issue?)

**Comment by asterite** [comment]
Umm... do you need it to be **any** enum, or one enum in particular?

**Comment by rmosolgo** [comment]
In my dreams ... any enum :)

**Comment by asterite** [comment]
Closed in favor of #2733

**Comment by rmosolgo** [comment]
Thanks, I'll keep an eye on it!
